### ISSUE TITLE
Bug - "still contains unknown values during apply" error

### ISSUE THREAD

**Issue by hila3000** [hashicorp/terraform#28852]
### Terraform Version

0.14.5

### Terraform Configuration Files
```

provider "google" {
  version     = "2.20.3"
  project     = var.host_project
  region      = var.region
  alias       = "orig"
}

provider "google-beta" {
  version     = "2.20.3"
  project     = var.host_project
  region      = var.region
  alias       = "orig"
}

provider "google" {
  version       = "2.20.3"
  project       = var.host_project
  region        = var.region
  access_token  = data.google_service_account_access_token.default.access_token
}

provider "google-beta" {
  version       = "2.20.3"
  project       = var.host_project
  region        = var.region
  access_token  = data.google_service_account_access_token.default.access_token
}

provider "kubernetes" {
  load_config_file              = false
  version                       = "1.8.1"  ### Tried also with 1.13.4 and 2.2.0, same behavior
  host                          = "https://${module.create_gke.gke_cluster_endpoint}"
  token                         = data.google_client_config.default.access_token
  cluster_ca_certificate        = base64decode(module.create_gke.gke_cluster_ca)
}

data "google_storage_bucket_object" "gitlab_domain_cer" {
  name   = "${var.env == "dev" ? "gitlab/gitlab-dev.cer" : "gitlab/gitlab.cer"}"
  bucket = "certs"
}

data "google_storage_bucket_object" "gitlab_domain_key" {
  name   = "${var.env == "dev" ? "gitlab/gitlab-dev.key" : "gitlab/gitlab.key"}"
  bucket = "certs"
}

data "http" "get_gitlab_domain_cer" {
  url = format("%s?alt=media", data.google_storage_bucket_object.gitlab_domain_cer.self_link)

  # Optional request headers
  request_headers = {
    Authorization = "Bearer ${data.google_client_config.default.access_token}"
  }
}

data "http" "get_gitlab_domain_key" {
  url = format("%s?alt=media", data.google_storage_bucket_object.gitlab_domain_key.self_link)

  # Optional request headers
  request_headers = {
    Authorization = "Bearer ${data.google_client_config.default.access_token}"
  }
}

resource "kubernetes_secret" "gitlab_domain_cer" {
  type = "kubernetes.io/tls"
  metadata {
    name = "gitlab-domain-cert"
    namespace = "gitlab"
  }
  data = {
    "tls.crt" = data.http.get_gitlab_domain_cer.body
    "tls.key" = data.http.get_gitlab_domain_key.body
  }
  depends_on = [data.http.get_gitlab_domain_cer, data.http.get_gitlab_domain_key]
}
```
Tried with and without the depends_on, same error

### Debug Output

Copying only relevant part of the debug-

2021/06/01 18:04:50 [TRACE] vertex "provider[\"registry.terraform.io/hashicorp/kubernetes\"]": visit complete
2021/06/01 18:04:50 [TRACE] dag/walk: visiting "kubernetes_secret.gitlab_domain_cer (expand)"
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer (expand)": starting visit (*terraform.nodeExpandApplyableResource)
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer (expand)": expanding dynamic subgraph
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer (expand)": entering dynamic subgraph
2021/06/01 18:04:50 [TRACE] dag/walk: visiting "kubernetes_secret.gitlab_domain_cer"
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer": starting visit (*terraform.NodeApplyableResource)
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer": visit complete
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer (expand)": dynamic subgraph completed successfully
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer (expand)": visit complete
2021/06/01 18:04:50 [TRACE] dag/walk: visiting "kubernetes_secret.gitlab_domain_cer"
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer": starting visit (*terraform.NodeApplyableResourceInstance)
2021/06/01 18:04:50 [TRACE] EvalReadDiff: Read Create change from plan for kubernetes_secret.gitlab_domain_cer
2021/06/01 18:04:50 [TRACE] EvalReadState: reading state for kubernetes_secret.gitlab_domain_cer
2021/06/01 18:04:50 [TRACE] EvalReadState: no state present for kubernetes_secret.gitlab_domain_cer
2021/06/01 18:04:50 [TRACE] EvalReadDiff: Read Create change from plan for kubernetes_secret.gitlab_domain_cer
2021/06/01 18:04:50 [TRACE] Re-validating config for "kubernetes_secret.gitlab_domain_cer"
2021/06/01 18:04:50 [TRACE] GRPCProvider: ValidateResourceTypeConfig
2021/06/01 18:04:50 [TRACE] GRPCProvider: PlanResourceChange
2021/06/01 18:04:50 [WARN] Provider "registry.terraform.io/hashicorp/kubernetes" produced an invalid plan for kubernetes_secret.gitlab_domain_cer, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .data: sensitive planned value does not match config value
2021/06/01 18:04:50 [TRACE] EvalCheckPlannedChange: Verifying that actual change (action Create) matches planned change (action Create)
2021/06/01 18:04:50 [TRACE] EvalReadState: reading state for kubernetes_secret.gitlab_domain_cer
2021/06/01 18:04:50 [TRACE] EvalReadState: no state present for kubernetes_secret.gitlab_domain_cer
kubernetes_secret.gitlab_domain_cer: Creating...
2021/06/01 18:04:50 [TRACE] vertex "kubernetes_secret.gitlab_domain_cer": visit complete
2021/06/01 18:04:50 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/kubernetes\"] (close)" errored, so skipping
2021/06/01 18:04:50 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2021/06/01 18:04:50 [TRACE] dag/walk: upstream of "root" errored, so skipping

Error: configuration for kubernetes_secret.gitlab_domain_cer still contains unknown values during apply (this is a bug in Terraform; please report it!)

2021-06-01T18:04:52.042+0300 [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/kubernetes/1.8.1/darwin_amd64/terraform-provider-kubernetes_v1.8.1_x4 pid=87798
2021-06-01T18:04:52.042+0300 [DEBUG] plugin: plugin exited

### Crash Output

Error: configuration for kubernetes_secret.gitlab_domain_cer still contains unknown values during apply (this is a bug in Terraform; please report it!)

### Expected Behavior

secret should get created successfully.

### Actual Behavior

secret wasn't created at all.

### Steps to Reproduce

Apply the above configuration on a GCP project

### Additional Context

We are running terraform in a wrapper script but I don't think it's related.

**Comment by jbardin** [comment]
Hi @hila3000,

Thanks for filing the issue. There have been numerous fixes in the 0.14 branch since the 0.14.5 release, and even more  in the 0.15 branch. Can you verify if this configuration still exhibits the problem with a current release, preferably v0.15.4 or at least 0.14.11?

Thanks!

**Comment by hila3000** [comment]
Yes, this issue also happens with 0.14.11 (can't really check 0.15.4 since it's a state for production environment, so need to stay in the 0.14.x range until we decide to migrate all projects to use 0.15.x)

**Comment by hila3000** [comment]
I tried also importing this to the state and then rerunning TF and still it fails, same error. plan shows-
`  # kubernetes_secret.gitlab_domain_cert will be updated in-place
  ~ resource "kubernetes_secret" "gitlab_domain_cer" {
      ~ data = (sensitive value)
        id   = "gitlab/gitlab-domain-cert"
        # (1 unchanged attribute hidden)

        # (1 unchanged block hidden)
    }
`

**Comment by jbardin** [comment]
Thanks @hila3000,

The only scenario I can think of is that one of the data sources is somehow causing an error that isn't being caught, leaving the unknown values in the state. The logs you provided are not complete, and don't really point to possible problems. The complete trace log may have more information that could indicate what is going on, but since this is a bug somewhere, upgrading to the latest version is going to be required to avoid it even if it still exists in the most recent release.

**Comment by hila3000** [comment]
You mean upgrade to latest 0.14.x (which I did) or to latest 0.15.x?

**Comment by jbardin** [comment]
Sorry, what I meant is that it may be fixed in 0.15 (which I believe is likely), but even if there is still bug to patch, there are not going to be any more 0.14 maintenance releases. This means reproducing and verifying the issue in 0.15 is the next step we need to take here. If the issue is resolved in 0.15 there is nothing more to do, and if it is not we can work with the current codebase to find a solution.

**Comment by hila3000** [comment]
OK then, so I'm trying to check it with 0.15.4. I've set it in required_version, but I get this error-
`Error: Unsupported Terraform Core version

  on versions.tf line 2, in terraform:
   2:   required_version = "0.15.4"

This configuration does not support Terraform version 0.14.5. To proceed,
either choose another supported Terraform version or update this version
constraint. Version constraints are normally set for good reason, so updating
the constraint may lead to other errors or unexpected behavior.`

Do I need to go through a different 0.15.x version prior of trying 0.15.4? (If so, it's not specified in the upgrade doc from v0.14 to v0.15)

**Comment by hila3000** [comment]
I checked with 0.15.4, and the issue is not reproduced there, a.k.a the secret is being created with no issues.
It's problematic for us to upgrade 0.15.4 cause we use TF in a wrapper and we already in a migration process to 0.14.5..  Is there a chance it could get fixed in any 0.14.x version?

**Comment by jbardin** [comment]
Thanks for confirming that @hila3000!

There shouldn't be any structural state changes between 0.14 and 0.15 to worry about here (which is the downgrade back to to 0.14 can now happen), but without knowing exactly what was causing it I cannot say for sure it won't reoccur. 

Do you have the ability to fetch prior versions of the state? I would be curious to see what changed when you did this upgrade which now causes it to work with 0.14 if possible (minus any sensitive data of course). Since this was most likely a bug fixed in 0.15 I'm going to close this out, but feel free to reply here if you have any other concerns. 

Thanks!

**Comment by hila3000** [comment]
I don't have ability to check prior versions cause this is a new project which I installed with 0.14.5 initially.
